TutorLizard.BusinessLogic/Services/TutorService.cs

```diff
@@ -14,39 +14,30 @@ public TutorService(ITutorDataAccess dataAccess, IUserIdentityService userIdenti
     }
     public Ad CreateAd(string subject, string title, string description)
     {
-        // return created Ad
-        // return empty with id = 0 to indicate something went wrong
-        return new Ad()
+        int? tutorId = _userIdentityService.GetUserId();
+        if (tutorId is null)
         {
-            // this is only for tests
-            Id = 1,
-            Subject = subject,
-            Title = title,
-            Description = description
-        };
+            return new Ad()
+            { Id = 0 };
+        }
+
+        return _dataAccess.CreateAd((int)tutorId, subject, title, description);
+
     }
     public ScheduleItem CreateScheduleItem(int adId, DateTime dateTime)
     {
-        // return created ScheduleItem
-        // return empty with id = 0 to indicate something went wrong
-        return new ScheduleItem()
-        {
-            // this is only for tests
-            Id = 1,
-            AdId = adId,
-            DateTime = dateTime
-        };
+        return _dataAccess.CreateScheduleItem(adId, dateTime);
     }
-    public Ad? GetAdById(int adId)
+    public Ad? GetAdById(int? adId)
     {
         // return Ad (from _dataAccess) with provided adId
         // return null if no such Ad
-        return new Ad()
+        adId = _userIdentityService.GetAllAdsId();
```

Comment:
You already have adId, because you received it as a parameter. Don't overwrite parameters.
Now you should either ask `_dataAccess` to give you an Ad? (nullable, because there could be none with such Id) or ask `_dataAccess` for list of all Ads, filter it to find matching Id and return what you find (null if nothing).